pkg/windows/calico.go

```diff
@@ -59,25 +59,52 @@ func (c *Calico) Setup(ctx context.Context, dataDir string, nodeConfig *daemonco
 	if err != nil {
 		return nil, err
 	}
-
-	secrets, err := client.CoreV1().Secrets(CalicoSystemNamespace).List(ctx, metav1.ListOptions{})
+	serviceAccounts := client.CoreV1().ServiceAccounts(CalicoSystemNamespace)
+	calicoSA, err := serviceAccounts.Get(ctx, calicoNode, metav1.GetOptions{})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("[Setup] failed to return %s service account: %v", calicoNode, err)
+	}
+	//sa, err := serviceaccount.GetOrCreateServiceAccount(client.CoreV1(), CalicoSystemNamespace, calicoNode)
+	//if err != nil {
+	//	return nil, fmt.Errorf("[Setup] failed to return service account: %v", err)
+	//}
+
+	t := true
+	calicoSA.AutomountServiceAccountToken = &t
+	nc := nodeConfig
+	var audiences []string
+	day := int64(3600 * 24)
+	req := authv1.TokenRequest{
+		TypeMeta:   calicoSA.TypeMeta,
+		ObjectMeta: calicoSA.ObjectMeta,
```

Comment:
I don't think this is going to work - you're creating a TokenRequest object but giving it the TypeMeta and ObjectMeta from the SA which is of a completely different kind. TypeMeta and ObjectMeta aren't normally directly assigned to. Looking at example code from upstream, I don't think you need to set these at all.
```suggestion
```